retry-on-conflict/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+)
+
+var (
+	namespace = "default"
+	name      = "foobar"
+)
+
+func main() {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+
+	cfg, err := clientcmd.BuildConfigFromFlags("", path.Join(home, ".kube/config"))
+	if err != nil {
+		panic(err.Error())
+	}
+
+	client := kubernetes.NewForConfigOrDie(cfg)
+	desired := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string]string{"foo": "bar"},
+	}
+
+	defer deleteConfigMap(client, name, namespace)
```

Comment:
I think deferring this call would make more sense after the successful creation.